File: 2020/ruby/src/day02.rb
module Day02
	class Day
		def part1(input)
			out = 0
			input.each {|line|
				spl = line.split(/[- :]+/)
				v = spl[-1].count(spl[-2])
				if spl[0].to_i<=v && v<=spl[1].to_i
					out+=1
				end
			}
			return out
		end

		def part2(input)
			out = 0
			input.each {|line|
				spl = line.split(/[- :]+/)
				if (spl[-1][spl[0].to_i-1]==spl[-2]) ^ (spl[-1][spl[1].to_i-1]==spl[-2])
					out+=1
				end
			}
			return out
		end

		def run(input)
			[part1(input), part2(input)]
		end
	end
end

File: 2020/python/src/day02.py
from utils.aocUtils import *
import re
def main(input:str):
	p1 = 0
	p2 = 0
	for l in input.splitlines():
		spl = re.split("[- :]+", l)
		if int(spl[0])<=spl[-1].count(spl[-2])<=int(spl[1]):
			p1+=1
		if (spl[-1][int(spl[0])-1] == spl[2]) != (spl[-1][int(spl[1])-1] == spl[2]):
			p2+=1
	return (p1, p2)